### ISSUE TITLE
Compatibility with Angular

### ISSUE THREAD

**Issue by sondreb** [TBD54566975/web5-js#173]
tldr: Add "skipLibCheck": true to your tsconfig.json.

I believe it to be important to support Angular out of the box without any special setup required for developers. There are a few issues needed to be done for the most basic new Angular 16 project.

```sh
ng new app
cd app
npm install @tbd54566975/web5
```

Then just import the package in `app.component.ts`
```ts
import { Web5 } from "@tbd54566975/web5";
```

This will result in multiple lines of this error:

```
Error: node_modules/@tbd54566975/dwn-sdk-js/dist/types/src/core/message-reply.d.ts:2:31 - error TS7016: Could not find a declaration file for module 'readable-stream'. 'C:/src/temp14/app/node_modules/@tbd54566975/dwn-sdk-js/node_modules/readable-stream/lib/ours/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/readable-stream` if it exists or add a new declaration (.d.ts) file containing `declare module 'readable-stream';`

2 import type { Readable } from 'readable-stream';
                                ~~~~~~~~~~~~~~~~~
```

This fix this issue, one option is to allow implicit any in the `tsconfig.json`:

```json
"noImplicitAny": false,
```

This then reduces the errors to a single one:

```
Error: node_modules/@tbd54566975/dwn-sdk-js/dist/types/src/store/level-wrapper.d.ts:1:23 - error TS1452: 'resolution-mode' assertions are only supported when `moduleResolution` is `node16` or `nodenext`.

1 /// <reference types="node" resolution-mode="require"/>
```

It's not a valid solution to change the moduleResolution to either of those options, this breaks Angular Core.

So one option to fix this issue is to add another configuration to `tsconfig.json`:

```json
"skipLibCheck": true,
```

This can also be used to avoid setting noImplicityAny, as it removes validation of the dependent libraries.

**Comment by sondreb** [comment]
With the latest 0.8.1 there are more issues and not possible to get it working easily with Angular. I will do some more attempts and likely it will require custom webpack configuration for it.

**Comment by sondreb** [comment]
Instructions to reproduce usage and issues:

1. Create an new project using Angular CLI, with latest angular and latest Node.js (20.x)
2. ` npm install @web5/api` and import the package in any Agnular component.
4. `npm start`

This will result in errors like the following:

![image](https://github.com/TBD54566975/web5-js/assets/309938/a01a29f5-c159-4ad0-b2c8-00db7a5ecb33)

5. `npm install crypto-browserify` and `npm install stream-browserify`
6. `npm start`

Now results in:
![image](https://github.com/TBD54566975/web5-js/assets/309938/3a095956-b385-4647-bdcc-7e1883fe7a55)

7. `npm i --save-dev @types/readable-stream`
8. `npm start`

Now results in:
![image](https://github.com/TBD54566975/web5-js/assets/309938/2315c952-f11b-4288-a29b-1acd42d53f72)

9. Next step is editing `tsconfig.json` by adding this to the compilerOptions section:

```
    "paths": {
      "crypto": ["node_modules/crypto-browserify"],
      "stream": ["node_modules/stream-browserify"]
    }
```
This reduces errors down to one, which sort of sounds like an actual TypeScript definition error in the package?

![image](https://github.com/TBD54566975/web5-js/assets/309938/b6a86176-08a8-4c69-bd81-f078b69f6a14)

This can be fixed by doing same as in OP:

"skipLibCheck": true, within the `tsconfig.json` compilerOptions.

This makes the project compile successfully, not now it fails at runtime:

![image](https://github.com/TBD54566975/web5-js/assets/309938/7abbe147-6663-49c9-9a69-5d03e094de31)

I have attempted various means of registering window as global in Angular, but this appears to have zero effect on this dependency and execution of the code in the screenshot.

**Comment by sondreb** [comment]
The global error can be fixed with this workaround:

```
 <script>
    if (global === undefined) {
      var global = window;
    }
  </script>
```

Within the index.html file.

Then another issue appears:
![image](https://github.com/TBD54566975/web5-js/assets/309938/13e6d7b4-18d4-4c1f-bf83-d6184e51b597)

By the sound of this error, I think the whole library believes to be running under Node.js context and not Web browser context? `process` is a Node.js native object.

**Comment by sondreb** [comment]
The "hack" to define global is not enough, the `process` library is required. So adding a polyfill.ts and configuring global, Buffer and process is required.

I made a repo that does all the required steps and the Web5 API now runs successfully on latest Angular:

https://github.com/sondreb/web5-angular
